### PR TITLE
Support modulus op

### PR DESCRIPTION
**Pull request by gangliao**
# Which issue does this PR close?

https://github.com/apache/arrow-rs/issues/98
https://github.com/apache/arrow-datafusion/issues/99
https://github.com/apache/arrow-rs/pull/317

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

no

### CODE REVIEW

**Comment by gangliao** [comment]
@alamb Is that possible to let `datafusion` use the latest code from `arrow-rs`?

**Comment by alamb** [comment]
Thank you for your contribution @gangliao !

> @alamb Is that possible to let datafusion use the latest code from arrow-rs?

As of now, we are only using "released" versions of arrow.rs. 

Your change in https://github.com/apache/arrow-rs/pull/317 is currently slated to be released in 4.3.0 (eta ~ 10 or so days) at which time it will be available for datafusion to use without any additional action


Until 4.3.0 is released, for your branch you can point datafusion to a sha from the arrow github to make sure the code works (e.g. how it was done prior to https://github.com/apache/arrow-datafusion/pull/395)

Basically we should wait until 4.3.0 is released and then this PR can be merged into datafusion

**Comment by gangliao** [comment]
Got it! Thank you for the clarification.

**Comment by alamb** [comment]
Arrow 4.3.0 has been released so if you rebase this PR it will likely be ready to go

**Comment by alamb** [comment]
I am not sure what has happened to this PR but the diff looks not right:
Screen Shot 2021-06-16 at 12 38 29 PM

Where I would expect to only see the changes from https://github.com/apache/arrow-datafusion/pull/510/commits/97fd9fff3353eff3d4fd4441c29cd91c19fce939

**Comment by gangliao** [comment]
Yeah. Looks strange. I will resubmit a new PR tonight.

**Comment by gangliao** [comment]
https://github.com/apache/arrow-datafusion/pull/577
